### ISSUE TITLE
Caching and Content-type headers for zipkin-UI assets

### ISSUE THREAD

**Issue by adriancole** [openzipkin/zipkin-java#116]
zipkin-ui assets include the following:

```
zipkin-ui/448c34a56d699c29117adc64c43affeb.woff2
zipkin-ui/614fad616d014daf5367e068505cad35.png
zipkin-ui/89889688147bd7575d6327160d64e760.svg
zipkin-ui/8b55a822e72b8fd5e2ee069236f2d797.png
zipkin-ui/app.min.css
zipkin-ui/app.min.css.map
zipkin-ui/app.min.js
zipkin-ui/app.min.js.map
zipkin-ui/e18bbf611f2a2e43afc071aa2f4e1512.ttf
zipkin-ui/f4769f9bdb7466be65088239c12046d1.eot
zipkin-ui/fa2772327f55d8198301fdb8bcfc8158.woff
zipkin-ui/index.html
```

### Caching
The assets that aren't hashed (ex index.html) should have a cache timeout of 1 day, to prevent being reloaded on each click. Others can have a longer timeout as version updates will change the names (ex. 1 year)

### Content-Type
For some reason, the content-type of png is returning with a charset `Content-Type:image/png;charset=UTF-8`. This doesn't make sense.

This is the same basic problem as https://github.com/openzipkin/zipkin/issues/1058

**Comment by adriancole** [comment]
cc @eirslett @gena01

**Comment by adriancole** [comment]
Note: we probably also want to hint caching of /config.json to avoid that reloading on each click

**Comment by adriancole** [comment]
For the sake of simplicity, I used a single property `zipkin.ui.cache-period` to control all assets, with a value of 1 day.  The `cache-period` key matches existing properties in spring-boot and has the same behavior.

**Comment by adriancole** [comment]
The UTF-8 qualifier on png, will be a pain to remove. It is set by a global filter `CharacterEncodingFilter`, which is in fact needed as javascript contains characters outside the normal ascii range.

**Comment by adriancole** [comment]
fix for the content-type problem: #118

**Comment by adriancole** [comment]
I arbitrarily set default of all ui resources to a day (which implies force reload from cache if you are deploying or configuring often than that). It is configurable, but no per-asset granularity.

There are a few a resources that don't include hashes: index.html, app.min*, config.json. So, we have a dance of either making one knob, or several. Thoughts?

**Comment by abesto** [comment]
I'm pretty sure we can make `app[.min].{js,css}[.map]` have a hash with `webpack`, in which case it'd be `index.html` (short cache period) vs everything else (infinite cache period), reducing the problem to just two knobs. Want?

**Comment by eirslett** [comment]
+1 for abesto's solution.

**Comment by adriancole** [comment]
SGTM!

**Comment by abesto** [comment]
Fixed by #117?
